src/mlj_model_interface.jl

```diff
@@ -165,15 +162,22 @@ function MLJModelInterface.update(model::MLJFluxModel,
     # Decide whether to enable entity embeddings (e.g., ImageClassifier won't)
     cat_inds = get_cat_inds(X)
     pure_continuous_input = (length(cat_inds) == 0)
-    enable_entity_embs = is_embedding_enabled_type(model) && !pure_continuous_input
+    enable_entity_embs = hasproperty(model, :embedding_dims) && !pure_continuous_input
```

Comment:
Why not use the trait as you had before? Testing existence of a field feels fragile to me.